src/simplify_rules.jl

```diff
@@ -132,20 +158,17 @@ let
     global serial_simplifier
     global serial_expand_simplifier
 
-    function default_simplifier(; kw...)
-        IfElse(has_trig_exp,
-               Postwalk(IfElse(x->symtype(x) <: Number,
-                               Chain((number_simplifier(),
-                                      trig_exp_simplifier())),
-                               If(x->symtype(x) <: Bool,
-                                  bool_simplifier()))
-                        ; kw...),
-               Postwalk(Chain((If(x->symtype(x) <: Number,
-                                  number_simplifier()),
-                               If(x->symtype(x) <: Bool,
-                                  bool_simplifier())))
-                        ; kw...))
-    end
+    default_simplifier(; kw...) = Postwalk(
+        Chain(
+            (
+                If(x->symtype(x) <: Number, number_simplifier()),
+                If(has_trig_exp, trig_exp_simplifier()),
```

Comment:
`has_*` could be expensive. Maybe we should have one combined `has_` function that returns `has_trig_exp, has_div_rem_mod` so there's only one traversal. This can be done in a later PR.